src/autocomplete.ts

```diff
@@ -40,6 +42,25 @@ export default class Autocomplete {
       this.feedback.setAttribute('aria-atomic', 'true')
     }
 
+    // if clearButton is not a button, show error
+    if (this.clearButton && this.clearButton.tagName.toLowerCase() !== 'button') {
+      const buttonErrorMessage = `Accessibility violation: ${this.clearButton.tagName} provided for clear button. Please use a "button" element.`
+      const buttonErrorText = '\u26a0 Error: See console'
+      const buttonError = document.createElement('span')
+      buttonError.setAttribute('style', 'color:#92140C')
+      buttonError.textContent = buttonErrorText
+      this.clearButton.parentNode?.replaceChild(buttonError, this.clearButton)
+      this.clearButton = buttonError
+      throw new Error(buttonErrorMessage)
+    }
+
+    // if clearButton doesn't have an accessible label, give it one
+    if (this.clearButton && !this.clearButton.getAttribute('aria-label')) {
+      const labelElem = document.querySelector(`label[for="${this.input.name}"]`)
+      const label = labelElem?.innerHTML || this.input.getAttribute('aria-label') || ''
```

Comment:
```suggestion
      const label = labelElem?.textContent || this.input.getAttribute('aria-label') || ''
```

We want to avoid injecting HTML into an attribute.

Reply:
Ahh nice catch. That would've been a fun problem.